### PR TITLE
Add progress button to Button guidelines page

### PR DESCRIPTION
**Pull request by mcarrano**
Adds progress button to button types section.

### CODE REVIEW

**Comment by mcarrano** [comment]
@mmenestr Added a new section for this under button types (https://patternfly-org-pr-2803-v4.surge.sh/v4/components/button/design-guidelines#progress-button). Please review.

File: packages/v4/src/content/design-guidelines/components/button/button.md [hunk 1]
```diff
@@ -99,6 +99,22 @@ Call To Action buttons and links are used on websites to direct users to complet
...
+
+<img src="./img/progress-button.png" alt="Example of progress button usage" width="1435"/>
+
+1. The user clicks the submit button.
+
+2. A spinner indicates that the operation is in progress.
+
+3. When complete, use a success icon (fa-check-circle) and a change in the label text to indicate that the operation was successfully completed.
+
+Use a progress button when clicking a submit button will not navigate the user to a new page or open a modal. For operations that will take more than a few seconds, use of a modal with a progress bar is recommended to reflect status and indicate time to completion.
```
Comment (mmenestr) [inline packages/v4/src/content/design-guidelines/components/button/button.md:116]: All looks great, my only comment is I would move this to be above the image, with the rest of the general guidelines around progress buttons!

**Comment by mcarrano** [comment]
@mmenestr I made these changes. Please take another look.

**Review by mmenestr — APPROVED** [review]
This looks great @mcarrano !! Merging

**Pull request merged**